Sources/SwerverCore/Server/ClientRequestHandler.swift

```diff
@@ -10,18 +10,18 @@ public class ClientRequestHandler {
     
     public func handle(client: Socket) throws {
         let httpRequest = try parseRequest(clientSocket: client)
-        try client.write(from: httpRouteProcessor.process(request: httpRequest))
+        try client.write(from: httpRouteProcessor.process(request: (httpRequest)))
     }
     
-    private func parseRequest(clientSocket: Socket) throws -> HttpRequest {
-        var readData: Data? = Data(capacity: Swerver.bufferSize)
-        let _ = try clientSocket.read(into: &readData!)
-        guard let parsedData = String(data: readData!, encoding: .utf8) else {
-            readData!.count = 0
+    private func parseRequest(clientSocket: Socket) throws -> HttpRequest? {
+        var readData: Data = Data(capacity: Swerver.bufferSize)
+        let _ = try clientSocket.read(into: &readData)
+        guard let parsedData = String(data: readData, encoding: .utf8) else {
```

Comment:
Given that there are two distinct branches here, I think an `if let` with an explicit `else` branch would be easier to read than a `guard`

Reply:
will look into that

Reply:
fixed